### PR TITLE
[docs] [rllib] Remove outdated RLlib examples from the docs 1/N

### PR DESCRIPTION
**Pull request by angelinalg**
Removing outdated examples.
cc: @sven1977 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-overview/examples.rst [hunk 1]
```diff
@@ -633,54 +633,12 @@ Ray Examples
...
-
-        VizDoom example script using RLlib's auto-attention wrapper
-
     .. grid-item-card:: :bdg-secondary:`Code example`
         :class-item: gallery-item rl
         :link: https://github.com/ray-project/ray/blob/master/rllib/env/tests/test_env_with_subprocess.py
 
         Example of how to ensure subprocesses spawned by envs are killed when RLlib exits.
 
-    .. grid-item-card:: :bdg-secondary:`Code example`
```
Comment (sven1977) [inline doc/source/ray-overview/examples.rst:678]: Actually, let's leave this one in. The example is still there and working.

File: doc/source/rllib/rllib-examples.rst [hunk 2]
```diff
@@ -43,27 +43,13 @@ Environments and Adapters
...
-- `VizDoom example script using RLlib's auto-attention wrapper <https://github.com/ray-project/ray/blob/master/rllib/examples/vizdoom_with_attention_net.py>`__:
-   Script showing how to run PPO with an attention net against a VizDoom gym environment.
 - `Subprocess environment <https://github.com/ray-project/ray/blob/master/rllib/env/tests/test_env_with_subprocess.py>`__:
    Example of how to ensure subprocesses spawned by envs are killed when RLlib exits.
 
 
 Custom- and Complex Models
 --------------------------
 
-- `Attention Net (GTrXL) learning the "repeat-after-me" environment <https://github.com/ray-project/ray/blob/master/rllib/examples/attention_net.py>`__:
```
Comment (sven1977) [inline doc/source/rllib/rllib-examples.rst:65]: Same here. This one should remain here.

File: doc/source/rllib/rllib-models.rst [hunk 3]
```diff
@@ -351,8 +351,7 @@ Check out RLlib's `GTrXL (Attention Net) <https://arxiv.org/abs/1910.06764>`__ i
 to get a better idea on how to write your own models of this type. These are the models we use
 as wrappers when ``use_attention=True``.
 
-You can run `this example script <https://github.com/ray-project/ray/blob/master/rllib/examples/attention_net.py>`__ to run these nets within some of our algorithms.
-`There is also a test case <https://github.com/ray-project/ray/blob/master/rllib/models/tests/test_attention_nets.py>`__, which confirms their learning capabilities in PPO and IMPALA.
```
Comment (sven1977) [inline doc/source/rllib/rllib-models.rst:355]: Same: Remain.

File: release/rllib_tests/unit_gpu_tests/run.sh [hunk 4]
```diff
@@ -22,9 +22,6 @@ SUCCESS=1
 # Run all test cases, but with a forced num_gpus=1.
 export RLLIB_NUM_GPUS=1
 
-if python rllib/examples/attention_net.py --as-test --stop-reward=20 --num-cpus=0; then SUCCESS=0; fi
```
Comment (sven1977) [inline release/rllib_tests/unit_gpu_tests/run.sh:25]: Same: Remain.

File: release/rllib_tests/unit_gpu_tests/run.sh [hunk 5]
```diff
@@ -22,9 +22,6 @@ SUCCESS=1
...
-if python rllib/examples/attention_net.py --framework=torch --as-test --stop-reward=20 --num-cpus=0; then SUCCESS=0; fi
```
Comment (sven1977) [inline release/rllib_tests/unit_gpu_tests/run.sh:26]: Same: Remain.

File: rllib/BUILD [hunk 6]
```diff
@@ -1884,24 +1884,6 @@ py_test(
     args = ["--stop-iter=2", "--framework=torch"]
 )
 
-py_test(
```
Comment (sven1977) [inline rllib/BUILD:1887]: Same: Remain.

File: rllib/examples/attention_net.py [hunk 7]
```diff
@@ -1,225 +0,0 @@
-"""
```
Comment (sven1977) [inline rllib/examples/attention_net.py:1]: Same: Let's leave this in.

**Review by sven1977 — APPROVED** [review]
LGTM! Thanks @angelinalg

File: rllib/BUILD (hunk 6 above)
Comment (sven1977) [inline rllib/BUILD:1887]: You have to also remove from this file: `test_env_with_subprocess`.

**Review by pcmoritz — APPROVED** [review]

**Pull request merged**